framework/postcommit_tasks/handlers.py

```diff
@@ -1,38 +1,41 @@
 import logging
 import threading
 
+import gevent
+
 from website import settings
 
 _local = threading.local()
 logger = logging.getLogger(__name__)
 
 def postcommit_queue():
     if not hasattr(_local, 'postcommit_queue'):
-        _local.postcommit_queue = []
+        _local.postcommit_queue = set()
     return _local.postcommit_queue
 
 def postcommit_before_request():
-    _local.postcommit_queue = []
+    _local.postcommit_queue = set()
 
 def postcommit_after_request(response, base_status_error_code=500):
     if response.status_code >= base_status_error_code:
-        _local.postcommmit_queue = []
+        _local.postcommmit_queue = set()
         return response
     try:
         if postcommit_queue():
-            for task in postcommit_queue():
-                task()
+            threads = [gevent.spawn(func, *args) for func, args in postcommit_queue()]
```

Comment:
I don't think order should ever matter. There's one URL for each object. The set is a list of every unique thing that was saved during the request so they should all be removed. 

It won't work without gevent but gevent is in the base requirements.txt now so it shouldn't need to.